lib/express.js

```diff
@@ -33,21 +57,26 @@ exports = module.exports = createApplication;
  * @api public
  */
 
-function createApplication() {
+function createApplication (opts) {
+  var options = opts || {}
+  options.reqProto = options.reqProto || http.IncomingMessage.prototype
+  options.resProto = options.resProto || http.ServerResponse.prototype
+  options.createServer = options.createServer || http.createServer
```

Comment:
Instead of making the `http` object above, why not just `require('http')` here instead? You'd avoid three requires. Instead, since all 3 of these options should be expected to be set together (why would you mix http native with implementation?), just make them all required or fallback on something like `var options = opts || getHttpOptions()`.